src/util/TokenRemover.ttslua

```diff
@@ -1,72 +1,80 @@
---- Created by Whimsical.
---- DateTime: 2021-02-02 9:41 a.m.
----
-
 local zone = nil
+local CHAOS_TOKEN_NAMES = {
+  "Elder Sign",
+  "+1",
+  "0",
+  "-1",
+  "-2",
+  "-3",
+  "-4",
+  "-5",
+  "-6",
+  "-7",
+  "-8",
+  "Skull",
+  "Cultist",
+  "Tablet",
+  "Elder Thing",
+  "Auto-fail",
+  "Bless",
+  "Curse",
+  "Frost"
+}
 
--- Forward Declaration
----@param is_enabled boolean
-local setMenu = function(is_enabled) end
+-- general code
+function onSave()
+  return JSON.encode(zone and zone.getGUID() or nil)
+end
 
-local function enable()
-    if self.held_by_color~=nil then return end
-    local position = self:getPosition()
-    local rotation = self:getRotation()
-    local scale = self:getScale()
+function onLoad(savedData)
+  if savedData ~= "" and savedData ~= nil then
+    zone = getObjectFromGUID(JSON.decode(savedData))
+  end
+  setMenu(zone == nil)
+end
 
-    zone = spawnObject {
-        type = "ScriptingTrigger",
-        position = Vector(position.x, position.y+25+(bit32.rshift(scale.y, 1))+0.41, position.z),
-        rotation = rotation,
-        scale = Vector(scale.x*2, 50, scale.z*2),
-        sound = true,
-        snap_to_grid = true
-    }
+-- context menu functions
+function enable()
+  local scale = self.getScale()
 
-    setMenu(false)
-end
+  zone = spawnObject({
+    type = "ScriptingTrigger",
+    position = self.getPosition() + Vector(0, 2.5 + 0.11, 0),
+    rotation = self.getRotation(),
+    scale = { scale.x * 2, 5, scale.z * 2 }
+  })
 
-local function disable()
-    if zone~=nil then zone:destruct() end
-    setMenu(true)
+  setMenu(false)
 end
 
----@param is_enabled boolean
-setMenu = function(is_enabled)
-    self:clearContextMenu()
-    if is_enabled then
-        self:addContextMenuItem("Enable", enable, false)
-    else
-        self:addContextMenuItem("Disable", disable, false)
-    end
+function disable()
+  if zone ~= nil then zone.destruct() end
+  setMenu(true)
 end
 
-function onLoad(save_state)
-    if save_state=="" then return end
-    local data = JSON.decode(save_state)
-    zone = getObjectFromGUID(data.zone)
-    setMenu(zone==nil)
+-- core functions
+function setMenu(isEnabled)
+  self.clearContextMenu()
+  if isEnabled then
+    self.addContextMenuItem("Enable", enable)
+  else
+    self.addContextMenuItem("Disable", disable)
+  end
 end
 
-function onSave()
-    return JSON.encode {
-        zone = zone and zone:getGUID() or nil
-    }
-end
+function onObjectEnterScriptingZone(entering, object)
+  if zone ~= entering or object == self or object.type == "Deck" or object.type == "Card" then
+    return
+  end
 
----@param entering TTSObject
----@param object TTSObject
-function onObjectEnterScriptingZone(entering , object)
-    if zone~=entering then return end
-    if object==self then return end
-    if object.type=="Deck" or object.type=="Card" then return end
+  local name = object.getName()
+  for _, entry in ipairs(CHAOS_TOKEN_NAMES) do
```

Comment:
Can make the token name list a map using the names as keys, will remove the loop here and give an O(1) check time

Reply:
Updated